tests/test_linux_pstore.py

```diff
@@ -1,6 +1,32 @@
+import re
+
+
 def test_pstore_fs(shell):
     """
     Test if the pstore filesystem exists.
     """
 
     shell.run_check("test -d /sys/fs/pstore")
+
+
+def test_kernel_messages(shell, check):
+    """
+    Test if the kernel only logs messages that we expect.
+    """
+
+    expected = [
+        re.compile(r"^\[\s*\d+\.\d+\] spi_stm32 44009000\.spi: failed to request (tx|rx) dma channel"),
+        re.compile(r"^\[\s*\d+\.\d+\] clk: failed to reparent ethck_k to pll4_p: -22"),
+        re.compile(r"^\[\s*\d+\.\d+\] stm32-dwmac 5800a000\.ethernet switch: Adding VLAN ID 0 is not supported"),
+    ]
+
+    messages = shell.run_check("dmesg -l warn -l err -l crit -l alert -l emerg -k")
+
+    # Check if we do not have more messages than expected
+    for message in messages:
+        for e in expected:
+            if e.match(message):
+                break
+        else:
+            with check:
+                raise AssertionError(f"Kernel warned '{message}', but we did not expect this.")
```

Comment:
I could have sworn we already had that somewhere, but I guess we didn't.

I think it would be great if we also caught these errors going away. Then we can remove them from the list.

How about doing the inverse of the current approach:

- Read each message into a `set()`. Use something like `re.sub(r"^\[\s*\d+\.\d+\]", "[    0.000000]")` to remove the variability of the timestamps.
- `assert messages_set == expected_messages_set`

Reply:
> I could have sworn we already had that somewhere, but I guess we didn't.

Me too. We had that on the roadmap some time last year. It seems we have lost that somewhere.

> I think it would be great if we also caught these errors going away. Then we can remove them from the list.

OK. Sounds reasonable.

> How about doing the inverse of the current approach:

I initially thought I would need a lot more regex to match the messages. But in the end there is only one message that benefits from the regexes here.
I would be fine to change it this way and use sets. This way we would see if messages go away or new ones pop up.